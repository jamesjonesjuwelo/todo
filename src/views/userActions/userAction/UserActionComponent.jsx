import React from 'react';
import styled from 'styled-components';

const userActionComponent = ({ userAction }) => {
  const { description, type } = userAction;
  
  return (
    <Container>
      <Description>{description}</Description>
      <Type>{type}</Type>
    </Container>
  
  )
};

const faintOrange = 'rgba(231, 111, 81, 0.3)';

const Container = styled.div`
  padding: 10px;
  position: relative;
  border-bottom: 1px solid ${faintOrange};
`;

const Description = styled.p`
  font-weight: bold;
`;

const Type = styled.p`
  color: grey;
  font-size: 12px;
`;

export default userActionComponent;



