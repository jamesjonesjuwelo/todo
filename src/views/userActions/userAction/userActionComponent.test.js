import React from 'react';
import { shallow } from 'enzyme';
import UserActionComponent from './UserActionComponent';
import 'jest-styled-components';

const props = {
  userAction: {
    description: 'Description',
    type: 'Type'
  }
};

describe('User Action Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( <UserActionComponent {...props} /> );
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  test(`contains props element`, () => {
    expect(wrapper.children().length).toBe(2);
  });
});