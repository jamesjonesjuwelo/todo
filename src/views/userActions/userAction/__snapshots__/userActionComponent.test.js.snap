// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Action Component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <userActionComponent
    userAction={
      Object {
        "description": "Description",
        "type": "Type",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.p>
          Description
        </styled.p>,
        <styled.p>
          Type
        </styled.p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Description",
        },
        "ref": null,
        "rendered": "Description",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Type",
        },
        "ref": null,
        "rendered": "Type",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.p>
            Description
          </styled.p>,
          <styled.p>
            Type
          </styled.p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Description",
          },
          "ref": null,
          "rendered": "Description",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Type",
          },
          "ref": null,
          "rendered": "Type",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
